spec/unit/array_spec.rb

```diff
@@ -35,28 +36,66 @@ class ArrayExampleArray < Machinery::Array
   }
 
   describe "#from_json" do
-    it "delegates to specialized class when the element class is set" do
-      json_object = [
-        { a: 1 },
-        { b: 2 }
-      ]
+    describe "plain arrays" do
+      it "delegates to specialized class when the element class is set" do
+        json_object = [
+          { a: 1 },
+          { b: 2 }
+        ]
 
-      array = ArrayExampleArray.from_json(json_object)
-      expect(array[0]).to eq(ArrayExampleObject.new(a: 1))
-      expect(array[1]).to eq(ArrayExampleObject.new(b: 2))
+        array = ArrayExampleArray.from_json(json_object)
+        expect(array[0]).to eq(ArrayExampleObject.new(a: 1))
+        expect(array[1]).to eq(ArrayExampleObject.new(b: 2))
+      end
+
+      it "uses generic classes when no element class is set" do
+        json_object = [1, { 2 => "2" }, [3, "3"]]
+
+        expected = Machinery::Array.new(
+          [
+            1,
+            Machinery::Object.new(2 => "2"),
+            Machinery::Array.new([3, "3"])
+          ]
+        )
+        expect(Machinery::Array.from_json(json_object)).to eq(expected)
+      end
     end
 
-    it "uses generic classes when no element class is set" do
-      json_object = [1, {2 => "2"}, [3, "3"]]
-
-      expected = Machinery::Array.new(
-        [
-          1,
-          Machinery::Object.new(2 => "2"),
-          Machinery::Array.new([3, "3"])
-        ]
-      )
-      expect(Machinery::Array.from_json(json_object)).to eq(expected)
+    describe "complex arrays" do
+      let(:array) {
+        json_object = {
+          "_attributes" => {
+          },
+          "_elements" => [
+            1,
+            2,
+            {
+              "_attributes" => {},
+              "_elements" => [1]
+            }
+          ]
+        }
+
+        Machinery::Array.from_json(json_object)
+      }
+
+      it "makes the elements available as the payload" do
+        expect(array.first).to eq(1)
+        expect(array[1]).to eq(2)
+        expect(array[2]).to eq(Machinery::Array.new([1]))
+      end
+
+      it "raises errors on unknown attributes" do
+        expect {
+          Machinery::Array.from_json({
+            "_attributes" => {
+              foo: "bar"
+            },
+            "_elements" => []
+          })
```

Comment:
Indent the right brace the same as the first position after the preceding left parenthesis.